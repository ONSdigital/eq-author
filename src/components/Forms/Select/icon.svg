<?xml version="1.0" encoding="UTF-8"?>
<svg width="7px" height="12px" viewBox="0 0 7 12" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="Artboard" fill="#D8D8D8">
            <polygon id="Triangle" points="3.46181818 0 6.92363636 4 0 4"></polygon>
            <polygon id="Triangle" transform="translate(3.461818, 10.000000) scale(1, -1) translate(-3.461818, -10.000000) " points="3.46181818 8 6.92363636 12 0 12"></polygon>
        </g>
    </g>
</svg>
