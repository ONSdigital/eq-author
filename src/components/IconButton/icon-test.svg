<?xml version="1.0" encoding="UTF-8"?>
<svg width="100px" height="95px" viewBox="0 0 100 95" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 46.2 (44496) - http://www.bohemiancoding.com/sketch -->
    <title>Artboard</title>
    <desc>Created with Sketch.</desc>
    <defs></defs>
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="Artboard" fill="#D8D8D8">
            <polygon id="Star" points="50 78 19.4351669 94.0688837 25.2725306 60.0344419 0.545061153 35.9311163 34.7175834 30.9655581 50 0 65.2824166 30.9655581 99.4549388 35.9311163 74.7274694 60.0344419 80.5648331 94.0688837"></polygon>
        </g>
    </g>
</svg>