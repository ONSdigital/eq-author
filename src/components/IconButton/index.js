import React from "react";
import styled, { css } from "styled-components";

import { colors } from "constants/theme";
import PropTypes from "prop-types";
import Tooltip from "components/Tooltip";
import VisuallyHidden from "components/VisuallyHidden";

const highlightOnHover = css`
  &:hover,
  &:focus {
    color: ${colors.highlight};
  }

  &:hover svg path,
  &:focus svg path {
    fill: ${colors.highlight};
  }
`;

const StyledButton = styled.button.attrs({
  type: "button"
})`
  background: none;
  border: none;
  display: flex;
  align-items: center;
  cursor: pointer;
  font-size: 0.875em;
  color: inherit;
  padding: 0;
  margin: 0;
  ${props => props.highlightOnHover && highlightOnHover};

  & svg {
    vertical-align: middle;
  }
`;

const Text = styled.span`
  margin-left: 0.6em;
`;

const withTooltip = (text, Component) => (
  <Tooltip content={text}>{Component}</Tooltip>
);

class IconButton extends React.Component {
  static propTypes = {
    icon: PropTypes.func.isRequired,
    iconOnly: PropTypes.bool,
    highlightOnHover: PropTypes.bool,
    children: PropTypes.node
  };

  static defaultProps = {
    highlightOnHover: true,
    iconOnly: false
  };

  render() {
    const { icon: Icon, children, iconOnly, ...otherProps } = this.props;

    const text = iconOnly ? (
      <VisuallyHidden>{children}</VisuallyHidden>
    ) : (
      <Text>{children}</Text>
    );

    const button = (
      <StyledButton iconOnly={iconOnly} {...otherProps}>
        <Icon />
        {text}
      </StyledButton>
    );

    return iconOnly ? withTooltip(children, button) : button;
  }
}

export default IconButton;
