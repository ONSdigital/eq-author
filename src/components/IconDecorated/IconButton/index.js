import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import IconDecorated from "components/IconDecorated";

export const Button = styled.button`
  display: flex;
  align-items: center;
  padding: 0.5em;
  cursor: pointer;
  appearance: none;
  border: none;
  opacity: 0.9;
  transition: opacity 200ms ease-out;
  background: transparent url(${props => props.icon}) no-repeat center;
  width: 3.5em;
  height: 3.5em;

  &:hover {
    opacity: 1;
  }

  &[disabled] {
    opacity: 0.5;
  }
`;

const IconButton = props => (
  <IconDecorated {...props} type="button" component={Button} />
);

IconButton.propTypes = {
  icon: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  handleClick: PropTypes.func,
  className: PropTypes.string
};

export default IconButton;
