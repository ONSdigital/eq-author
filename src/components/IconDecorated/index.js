/* eslint-disable react/forbid-prop-types */
import React from "react";
import PropTypes from "prop-types";
import Tooltip from "components/Tooltip/index";
import VisuallyHidden from "components/VisuallyHidden/index";

const IconDecorated = ({
  icon,
  title,
  disabled,
  handleClick,
  component: Component,
  ...otherProps
}) => (
  <Tooltip content={title}>
    <div>
      <Component
        onClick={handleClick}
        disabled={disabled}
        icon={icon}
        {...otherProps}
      >
        <VisuallyHidden>{title}</VisuallyHidden>
      </Component>
    </div>
  </Tooltip>
);

IconDecorated.propTypes = {
  icon: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  handleClick: PropTypes.func,
  className: PropTypes.string,
  component: PropTypes.any.isRequired
};

export default IconDecorated;
