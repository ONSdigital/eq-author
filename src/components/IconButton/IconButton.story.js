import React from "react";
import { storiesOf } from "@storybook/react";
import styled from "styled-components";
import IconButton from "components/IconButton";
import { action } from "@storybook/addon-actions";
import icon from "./test-icon.svg?inline";

const Wrapper = styled.div`
  padding: 1em;
  align-items: center;
`;

storiesOf("IconButton", module)
  .add("Single", () => (
    <Wrapper>
      <IconButton icon={icon} onClick={action("Click")}>
        Lorem ipsum
      </IconButton>
    </Wrapper>
  ))
  .add("IconOnly", () => (
    <Wrapper>
      <IconButton iconOnly icon={icon} onClick={action("Click")}>
        Lorem ipsum
      </IconButton>
    </Wrapper>
  ));
