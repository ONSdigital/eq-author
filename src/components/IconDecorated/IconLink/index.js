import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import IconDecorated from "components/IconDecorated";

export const StyledLink = styled.a`
  display: flex;
  align-items: center;
  padding: 0.5em;
  cursor: pointer;
  appearance: none;
  border: none;
  opacity: 0.9;
  transition: opacity 200ms ease-out;
  background: transparent url(${props => props.icon}) no-repeat center;
  width: 3.5em;
  height: 3.5em;

  &:hover {
    opacity: 1;
  }

  &[disabled] {
    opacity: 0.5;
  }
`;

const IconLink = props => <IconDecorated {...props} component={StyledLink} />;

IconLink.propTypes = {
  icon: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  handleClick: PropTypes.func,
  className: PropTypes.string
};

export default IconLink;
