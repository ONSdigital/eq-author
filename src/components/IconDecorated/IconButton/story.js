import React from "react";
import { storiesOf } from "@storybook/react";
import IconButton from "components/IconDecorated/IconButton/index";
import { colors } from "constants/theme";
import styled from "styled-components";
import icon from "./icon-test.svg";

const Background = styled.span`
  background-color: ${colors.darkGrey};
  padding: 1em;
  display: inline-block;
`;

storiesOf("IconButton", module)
  .addDecorator(story => <Background>{story()}</Background>)
  .add("Preview", () => <IconButton icon={icon} title="Preview" disabled />)
  .add("Export", () => <IconButton icon={icon} title="Export" disabled />);
