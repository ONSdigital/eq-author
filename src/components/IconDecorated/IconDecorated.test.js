import React from "react";
import { shallow } from "enzyme";
import IconDecorated from "components/IconDecorated";
import icon from "components/IconDecorated/IconLink/icon-test.svg";
import { Button } from "components/IconDecorated/IconButton";

describe("IconDecorated", () => {
  let wrapper;
  let handleClick;
  beforeEach(() => {
    handleClick = jest.fn();

    wrapper = shallow(
      <IconDecorated
        icon={icon}
        title="Title"
        type="button"
        component={Button}
        handleClick={handleClick}
      />
    );
  });

  it("should render a button", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("will handleClick", function() {
    wrapper.find(Button).simulate("click");
    expect(handleClick).toHaveBeenCalled();
  });
});
