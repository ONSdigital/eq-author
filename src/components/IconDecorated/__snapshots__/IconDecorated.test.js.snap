// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconDecorated should render a button 1`] = `
<Tooltip
  content="Title"
>
  <div>
    <IconButton__Button
      icon="icon-test.svg"
      onClick={[Function]}
      type="button"
    >
      <VisuallyHidden>
        Title
      </VisuallyHidden>
    </IconButton__Button>
  </div>
</Tooltip>
`;
